roxctl/netpol/connectivity/diff/netpol_test.go

```diff
@@ -0,0 +1,249 @@
+package diff
+
+import (
+	"errors"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/roxctl/common/environment/mocks"
+	"github.com/stackrox/rox/roxctl/common/npg"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestDiffAnalyzeNetpolCommand(t *testing.T) {
+	t.Parallel()
+	suite.Run(t, new(diffAnalyzeNetpolTestSuite))
+}
+
+type diffAnalyzeNetpolTestSuite struct {
+	suite.Suite
+}
+
+func (d *diffAnalyzeNetpolTestSuite) TestDiffAnalyzeNetpol() {
+	tmpOutFileName := d.T().TempDir() + "/out"
+	outFileTxt := tmpOutFileName + ".txt"
+	cases := []struct {
+		name                  string
+		inputFolderPath1      string
+		inputFolderPath2      string
+		strict                bool
+		stopOnFirstErr        bool
+		outFile               string
+		outputToFile          bool
+		outputFormat          string
+		removeOutputPath      bool
+		errStringContainment  bool
+		expectedValidateError error
+		expectedAnalysisError error
+	}{
+		{
+			name:                  "Empty dir1 input should return validation error",
+			inputFolderPath1:      "",
+			inputFolderPath2:      "",
+			expectedValidateError: errox.InvalidArgs,
+		},
+		{
+			name:                  "Empty dir2 should return validation error",
+			inputFolderPath1:      "/dev/null",
+			inputFolderPath2:      "",
+			expectedValidateError: errox.InvalidArgs,
+		},
+		{
+			name:                  "Not existing input folder paths should result in error 'os.ErrNotExist'",
+			inputFolderPath1:      "/tmp/xxx",
+			inputFolderPath2:      "/tmp/xxx",
+			expectedValidateError: nil,
+			expectedAnalysisError: os.ErrNotExist,
+		},
+		{
+			name:                  "Inputs with no resources errors should result in general NP-Guard error",
+			inputFolderPath1:      "testdata/empty-yamls",
+			inputFolderPath2:      "testdata/empty-yamls",
+			expectedValidateError: nil,
+			expectedAnalysisError: npg.ErrErrors,
+		},
+		{
+			name:                  "Treating warnings as errors should result in error of type 'npg.ErrWarnings'",
+			inputFolderPath1:      "testdata/acs-zeroday-with-invalid-doc",
+			inputFolderPath2:      "testdata/acs-zeroday-with-invalid-doc",
+			expectedAnalysisError: npg.ErrWarnings,
+			strict:                true,
+		},
+		{
+			name:                  "warnings not indicated without strict",
```

Comment:
I tried to rewrite the test case name into form `Doing A should result in B` but I failed here. Could we improve the names for the test cases in this file?